python/nano/src/bigdl/nano/pytorch/onnx/onnxrt_inference.py

```diff
@@ -184,6 +188,66 @@ def inference(self,
 # on_fit_start (LightningModule method overwrite)
 def on_fit_start(self):
     self._ortsess_up_to_date = False
+    self.exit_onnx()
+
+
+def train(self, mode=True):
+    self.exit_onnx()
+    self._train_mode(mode)
+
+
+def _forward_onnx(self, *args):
+    ort_inputs = {}
+    for i, ort_input_item in enumerate(args):
+        if isinstance(ort_input_item, torch.Tensor):
+            ort_input_item = ort_input_item.numpy()
+        ort_inputs[self._forward_args[i]] = ort_input_item
+    ort_outs = self._ortsess.run(None, ort_inputs)
+    return torch.from_numpy(ort_outs[0])
+
+
+def eval_onnx(self, input_sample=None, file_path="model.onnx", sess_options=None, **kwargs):
+    '''
+    This method change the `forward` method to an onnxruntime backed forwarding.
+
+    >>> model.eval_onnx()
+    >>> pred = model(x)  # onnxruntime forwarding
+    >>> model.exit_onnx()
+
+    :param input_sample: (optional) a torch dataloader, torch.Tensor or a
+           list of them for the model tracing.
+    :param file_path: (optional) The path to save onnx model file.
+    :param sess_options: (optional) ortsess options in ort.SessionOptions type.
+    :param **kwargs: (optional) will be passed to torch.onnx.export function.
+    '''
+    # change to eval mode
+    self.eval()
+
+    # get input_sample
+    if isinstance(input_sample, DataLoader):
+        input_sample = tuple(next(iter(input_sample))[:-1])
+    if input_sample is None and self.example_input_array:
+        input_sample = self.example_input_array
+    if input_sample is None and self.trainer is None:
+        raise RuntimeError("You must state an input_sample or fit on the model to use `eval_onnx`.")
```

Comment:
You must specify an input_sample or call Trainer.fit on the model first to use `eval_onnx`